### PR TITLE
update GPTQConfig and UTs

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

enhancement

## Description

part of https://github.com/intel/neural-compressor/pull/1549

## Expected Behavior & Potential Risk

UT pass

### CODE REVIEW

File: test/3x/torch/quantization/weight_only/test_gptq.py [hunk 1]
```diff
@@ -1,295 +1,183 @@
...
+        # test_default_rtn_config
+        model = copy.deepcopy(self.tiny_gptj)
+        quant_config = get_default_rtn_config()
+        model = quantize(model, quant_config, run_fn=run_fn)
+        rtn_label = model(self.example_inputs)[0]
+        rtn_atol = (rtn_label - self.label).amax()
+        # test_default_gptq_config
+        model = copy.deepcopy(self.tiny_gptj)
+        quant_config = get_default_gptq_config()
+        model = quantize(model, quant_config, run_fn=run_fn)
```
Comment (yiliu30) [inline test/3x/torch/quantization/weight_only/test_gptq.py:41]: I'm a little confused about using GPTQ with a custom dataloader, can you please provide an end-to-end example?

File: test/3x/torch/quantization/weight_only/test_gptq.py [hunk 2]
```diff
@@ -1,295 +1,183 @@
...
-        logger.info(f"Test GPTQ with config {quant_config}")
-        q_model = quantize(
-            model=model_1, quant_config=quant_config, run_fn=run_fn_for_gptq, run_args=dataloader_for_calibration
+        model = quantize(model, quant_config, run_fn=run_fn)
+        out = model(self.example_inputs)[0]
+        atol_true = (out - self.label).amax()
+        # compare atol, this case is an ideal case.
+        assert (
+            atol_false > atol_true
+        ), "use_mse_search=True doesn't help accuracy, maybe is reasonable, please double check."
```
Comment (yiliu30) [inline test/3x/torch/quantization/weight_only/test_gptq.py:95]: The outcome appears to be random; please avoid using assert or manually specifying a dedicated seed to ensure reproducibility.

File: test/3x/torch/quantization/weight_only/test_gptq.py [hunk 3]
```diff
@@ -1,295 +1,183 @@
...
+
+    # def test_layer_wise(self):
+    #     model = copy.deepcopy(self.tiny_gptj)
+    #     quant_config = GPTQConfig(
+    #         use_layer_wise=True,
+    #     )
+    #     model = quantize(model, quant_config, run_fn=run_fn)
+    # TODO(Xin): not implemented
```
Comment (yiliu30) [inline test/3x/torch/quantization/weight_only/test_gptq.py:103]:
```suggestion
    # TODO: (Xin) not implemented
```
Please use `TODO: ` as a keyword for the to-do comment.

File: neural_compressor/torch/quantization/config.py [hunk 4]
```diff
@@ -69,8 +69,11 @@ class RTNConfig(BaseConfig):
         "use_sym",
         "use_full_range",
         "use_mse_search",
-        "use_layer_wise",
         "export_compressed_model",
+        # layer wise params
+        "use_layer_wise",
+        "model_path",
+        # double quant
```
Comment (yiliu30) [inline neural_compressor/torch/quantization/config.py:76]: The arguments have been modified, please update the usage in the examples [examples](https://github.com/intel/neural-compressor/tree/master/examples/3.x_api/pytorch/nlp/huggingface_models/language-modeling/quantization/llm) accordingly. It is ok to do it in a separate PR. 
We would better create a JIRA to track it.

File: neural_compressor/torch/algorithms/layer_wise/torch_load.py [hunk 5]
```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/env python
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/layer_wise/torch_load.py:1]: The file located in the torch folder does not require the `torch_` prefix; it would be better to rename it to `load.py`.

File: test/3x/torch/quantization/weight_only/test_gptq.py (hunk 1 above)
Comment (xin3he) [reply test/3x/torch/quantization/weight_only/test_gptq.py:41]:
```
dataloder = Dataloader()
def run_fn(model):
    for data, label in dataloader:
        model(data)
```
The dataloader is defined/imported in the same file as run_fn then python can detect it.

File: neural_compressor/torch/algorithms/layer_wise/torch_load.py (hunk 5 above)
Comment (xin3he) [reply neural_compressor/torch/algorithms/layer_wise/torch_load.py:1]: ok, the layer_wise is not refined in this PR. It will come in the next PR of layer-wise support.

File: neural_compressor/torch/quantization/config.py (hunk 4 above)
Comment (xin3he) [reply neural_compressor/torch/quantization/config.py:76]: Got it, thanks.

File: test/3x/torch/quantization/weight_only/test_gptq.py (hunk 3 above)
Comment (xin3he) [reply test/3x/torch/quantization/weight_only/test_gptq.py:103]: OK

File: test/3x/torch/quantization/weight_only/test_gptq.py (hunk 2 above)
Comment (xin3he) [reply test/3x/torch/quantization/weight_only/test_gptq.py:95]: I think there is no random issue in UT. The assertion should always pass as currently checked or there may be a bug.

File: neural_compressor/torch/algorithms/weight_only/utility.py [hunk 6]
```diff
@@ -73,7 +73,7 @@ def quantize_4bit(tensor, quantile=1.0, dtype="nf4", return_int=False, **kwargs)
     mid_data = [(allow_data[i] + allow_data[i + 1]) / 2 for i in range(len(allow_data) - 1)]
     q_tensor = torch.zeros_like(tensor)
     for i in range(len(allow_data)):
-        data = allow_data_bit[i] if return_int else allow_data[i]
+        data = allow_data_bit[i] if return_int or "cast_int" in kwargs else allow_data[i]
```
Comment (Kaihui-intel) [inline neural_compressor/torch/algorithms/weight_only/utility.py:76]: What is the difference between `return_int` and `export_compressed_model`? We should unify all names similar to these.

**Review by Kaihui-intel — APPROVED** [review]

**Review by yiliu30 — APPROVED** [review]

**Review by YIYANGCAI — APPROVED** [review]

**Comment by chensuyue** [comment]
Example test failed, will fix in another PR.
